pkg/virt-launcher/virtwrap/manager_test.go

```diff
@@ -1622,6 +1624,95 @@ var _ = Describe("Manager", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(domSpec).ToNot(BeNil())
 		})
+
+		Context("on call to GetGuestOSInfo", func() {
+			var libvirtmanager DomainManager
+			var agentStore agentpoller.AsyncAgentStore
+
+			BeforeEach(func() {
+				agentStore = agentpoller.NewAsyncAgentStore()
+				libvirtmanager, _ = NewLibvirtDomainManager(mockConn, testVirtShareDir, nil, 0, &agentStore, "/usr/share/OVMF")
+			})
+
+			It("should return nil when no osInfo exists in the cache", func() {
+				osInfo := libvirtmanager.GetGuestOSInfo()
+				Expect(osInfo).To(BeNil())
+			})
+
+			It("should return osInfo when osInfo exists in the cache", func() {
+				fakeInfo := api.GuestOSInfo{
+					Name: "TestGuestOSName",
+				}
+
+				agentStore.Store(agentpoller.GET_OSINFO, fakeInfo)
+				osInfo := libvirtmanager.GetGuestOSInfo()
+				Expect(reflect.DeepEqual(&fakeInfo, osInfo)).To(BeTrue())
```

Comment:
Why do you need to use `reflect` here? Isn't a simple struct equality enough?

Reply:
Its a pointer, i can deref instead, but DeepEqual does it
wdyt?
lets first answer if we keep it as pointer (see please https://github.com/kubevirt/kubevirt/pull/5560#discussion_r629074725)